src/v/kafka/server/replicated_partition.h

```diff
@@ -66,6 +66,21 @@ class replicated_partition final : public kafka::partition_proxy::impl {
         return _translator->from_log_offset(_partition->high_watermark());
     }
 
+    model::offset log_end_offset() const {
+        return model::next_offset(log_dirty_offset());
+    }
+
+    model::offset log_dirty_offset() const {
+        if (_partition->is_read_replica_mode_enabled()) {
+            if (_partition->cloud_data_available()) {
+                return _partition->next_cloud_offset();
```

Comment:
wow. why is this necessary? feels like we have abstraction leaks.

Reply:
Agree, it is definitelly an abstraction leak. I used the same pattern as for the `high_watermark` it looks that the cloud offset doesn't have to be translated. It would be great to have it encapsulated in `cluster::partition`